src/buildercore/cfngen.py

```diff
@@ -484,10 +491,34 @@ def apply_delta(template, delta):
         for title in delta.minus[component]:
             del template[component][title]
 
-def regenerate_stack(pname, **more_context):
-    current_template = bootstrap.current_template(more_context['stackname'])
-    current_context = context_handler.load_context(more_context['stackname'])
-    write_template(more_context['stackname'], json.dumps(current_template))
-    context = build_context(pname, existing_context=current_context, **more_context)
-    delta = template_delta(pname, context)
-    return context, delta, current_context
+def regenerate_stack_vars(stackname, **more_context):
+    """returns the current template context and the new context for the given stackname.
+    use `cfngen.template_delta` to generate a list of changes"""
+    # fetch context used to build current stack
+    current_context = context_handler.load_context(stackname)
+    # don't rely on 'alt-config' being present in the current context, or if it is present,
+    # don't assume that alt-config existed when the current context existed. for example:
+    # `prod` used a local db before, same as default, but now a `prod` alt-config gets RDS
+    more_context['alt-config'] = choose_alt_config(stackname)
+    # build the context again, but this time re-use some current values/config
+    more_context['stackname'] = stackname # TODO: purge this crap
+    pname = core.parse_stackname(stackname)[0]
+    return current_context, build_context(pname, existing_context=current_context, **more_context)
+
+# def regenerate_stack(pname, **more_context):
+#    current_template = bootstrap.current_template(more_context['stackname'])
+#    current_context = context_handler.load_context(more_context['stackname'])
+#    write_template(more_context['stackname'], json.dumps(current_template))
+#    context = build_context(pname, existing_context=current_context, **more_context)
+#    delta = template_delta(pname, context)
+#    return context, delta, current_context
+
+def regenerate_stack(stackname, **more_context):
+    # what was the point of these two lines? it downloads the template body and saves it to disk and never uses it ...
```

Comment:
The point of downloading the template is that `template_delta` has to generate a diff (delta) to show it to the user (as updating the template is a dangerous operation). To generate the delta, it needs to get the template from the authoritative source (the CloudFormation API) and then compare it with the new one. The filesystem is acting as a light caching layer for the template as provided by the API, so it may just be a simplification that is worth the performance trade-off. Stacks are unfrequently updated.

Reply:
`template_delta` is using `read_template` which only looks at the filesystem at the moment, for this to work it would need to be modified to download it.